From 3bf0698bfa3357a066267e48548bfb555ce55d8e Mon Sep 17 00:00:00 2001
From: Marcelo Leitner <mleitner@redhat.com>
Date: Tue, 21 Oct 2014 16:16:54 -0400
Subject: [net] netfilter: nft_ct: add missing ifdef for NFT_MARK setting

Message-id: <20141021161654.22888.41046.stgit@localhost.localdomain>
Patchwork-id: 97923
O-Subject: [RHEL7 net PATCH 028/124] netfilter: nft_ct: add missing ifdef for NFT_MARK setting
Bugzilla: 1148041 1155088
RH-Acked-by: Florian Westphal <fwestpha@redhat.com>

Backport of upstream commit: e88e514e1fde119b567ff15b215d58b93722697c

 commit e88e514e1fde119b567ff15b215d58b93722697c
 Author: Patrick McHardy <kaber@trash.net>
 Date:   Sat Mar 29 10:43:01 2014 +0000

    netfilter: nft_ct: add missing ifdef for NFT_MARK setting

    The set operation for ct mark is only valid if CONFIG_NF_CONNTRACK_MARK is
    enabled.

    Signed-off-by: Patrick McHardy <kaber@trash.net>
    Signed-off-by: Pablo Neira Ayuso <pablo@netfilter.org>

Signed-off-by: Marcelo Ricardo Leitner <mleitner@redhat.com>
Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/netfilter/nft_ct.c b/net/netfilter/nft_ct.c
index bd0d41e..a2c45bd 100644
--- a/net/netfilter/nft_ct.c
+++ b/net/netfilter/nft_ct.c
@@ -268,8 +268,10 @@ static int nft_ct_init_validate_get(const struct nft_expr *expr,
 static int nft_ct_init_validate_set(uint32_t key)
 {
  switch (key) {
+#ifdef CONFIG_NF_CONNTRACK_MARK
  case NFT_CT_MARK:
   break;
+#endif
  default:
   return -EOPNOTSUPP;
  }
-- 
1.7.1